libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo/ast/internal/SchemaValidationScope.kt

```diff
@@ -326,7 +326,7 @@ private fun List<GQLSchemaExtension>.getImports(
          * Validate `@link` using a very minimal schema.
          * This ensure we can safely cast the arguments below
          */
-        val minimalSchema = builtinDefinitions + linkDefinitions()
+        val minimalSchema = builtinDefinitions
```

Comment:
Can we have `getImports` return its own foreign schema for `@link`? `@link` is a very special case and I'd rather keep everything contained in `getImports()`

Reply:
Something along these lines:

```
      if (gqlDirective.name == "link") {
        val linkForeignSchema = ForeignSchema("link", "v1.0", """
              directive @link(
                url: String!,  
                as: String,
                import: [Import],
                for: Purpose)
                repeatable on SCHEMA

              scalar Import

              enum Purpose {
                SECURITY
                EXECUTION
              }

            """.trimIndent().toGQLDocument().definitions
        )
        imports.add(
            Import(linkForeignSchema, linkForeignSchema.definitions, mapOf("link" to "link"))
        )
```